### PR TITLE
Fix child transmission target can't inherit the property on parent transmission

### PR DESCRIPTION
**Pull request by zhangcc01**
…ansmission target

Things to consider before you submit the PR:

* [ ] Has `CHANGELOG.md` been updated?
* [x] Are tests passing locally?
* [x] Are the files formatted correctly?
* [ ] Did you add unit tests if this modifies the Windows code?
* [x] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description

Open Unity Demo app
Select any Startup mode and restart demo app
Click Transmission target enter Event Name
Click Add Property button to add property "A" for Parent target
Then click Add Property button to add property "B" for Child target
Observe the result on Aria portal if the property "A" can be inherited to Child target

## Related PRs or issues

https://msmobilecenter.visualstudio.com/Mobile-Center/_workitems/edit/69752

## Misc

Add what's missing, notes on what you tested, additional thoughts or questions.

### CODE REVIEW

File: AppCenterDemoApp/Assets/Puppet/PuppetTransmission.cs [hunk 1]
```diff
@@ -446,9 +450,17 @@ public void TrackEventStringPropertiesChildTransmission()
             else
             {
                 var propertyConfigurator = transmissionTarget.GetPropertyConfigurator();
-                foreach (var property in properties)
-                {
-                    propertyConfigurator.SetEventProperty(property.Key, property.Value);
+                if (properties != null) {
+                  foreach (var property in properties)
```
Comment (annakocheshkova) [inline AppCenterDemoApp/Assets/Puppet/PuppetTransmission.cs]: Please fix indentation (it should be 4 spaces)

File: AppCenterDemoApp/Assets/Puppet/PuppetTransmission.cs [hunk 2]
```diff
@@ -487,7 +503,12 @@ public void TrackEventTypedPropertiesChildTransmission()
             else
             {
                 var propertyConfigurator = transmissionTarget.GetPropertyConfigurator();
-                PropertiesHelper.AddPropertiesToPropertyConfigurator(EventChildPropertiesList, propertyConfigurator);
+                if (properties != null) {
```
Comment (annakocheshkova) [inline AppCenterDemoApp/Assets/Puppet/PuppetTransmission.cs]: The opening bracket is on the next line:
```
if (...)
{
}
```

File: AppCenterDemoApp/Assets/Puppet/PuppetTransmission.cs [hunk 3]
```diff
@@ -487,7 +503,12 @@ public void TrackEventTypedPropertiesChildTransmission()
...
+                    PropertiesHelper.AddPropertiesToPropertyConfigurator(EventChildPropertiesList, propertyConfigurator);
+                }
+                if (propertiesParents != null) {
```
Comment (annakocheshkova) [inline AppCenterDemoApp/Assets/Puppet/PuppetTransmission.cs]: Same here

File: AppCenterDemoApp/Assets/Puppet/PuppetTransmission.cs [hunk 4]
```diff
@@ -446,9 +450,17 @@ public void TrackEventStringPropertiesChildTransmission()
...
-                foreach (var property in properties)
-                {
-                    propertyConfigurator.SetEventProperty(property.Key, property.Value);
+                if (properties != null) {
+                  foreach (var property in properties)
+                  {
+                      propertyConfigurator.SetEventProperty(property.Key, property.Value);
+                  }
+                }
+                if (propertiesParents != null) {
```
Comment (annakocheshkova) [inline AppCenterDemoApp/Assets/Puppet/PuppetTransmission.cs]: Same here

File: AppCenterDemoApp/Assets/Puppet/PuppetTransmission.cs (hunk 1 above)
Comment (zhangcc01) [reply AppCenterDemoApp/Assets/Puppet/PuppetTransmission.cs]: Update it in new commit.

**Review by annakocheshkova — APPROVED** [review]

File: AppCenterDemoApp/Assets/Puppet/PuppetTransmission.cs [hunk 5]
```diff
@@ -446,9 +450,19 @@ public void TrackEventStringPropertiesChildTransmission()
...
                 {
-                    propertyConfigurator.SetEventProperty(property.Key, property.Value);
+                    foreach (var property in properties)
+                    {
+                        propertyConfigurator.SetEventProperty(property.Key, property.Value);
+                    }
+                }
+                if (propertiesParents != null)
+                {
+                    foreach (var property in propertiesParents)
```
Comment (guperrot) [inline AppCenterDemoApp/Assets/Puppet/PuppetTransmission.cs:462]: 1DS supports inheriting from parent, here you are doing it in the app and not using the SDK mechanism.

Thus I don't really understand these changes. What am I missing?

**Review by guperrot — CHANGES_REQUESTED** [review]
See https://github.com/microsoft/appcenter-sdk-unity/pull/371#pullrequestreview-320220618

File: AppCenterDemoApp/Assets/Puppet/PuppetTransmission.cs (hunk 5 above)
Comment (guperrot) [reply AppCenterDemoApp/Assets/Puppet/PuppetTransmission.cs:462]: After talking to @annakocheshkova we believe this is not the right fix.
Native SDK has inheritance built in, please make sure that instead the child target is correctly created from parent and inheriting.

**Comment by annakocheshkova** [comment]
Closing this one, found the underlying problem #373

**Pull request closed without merging**